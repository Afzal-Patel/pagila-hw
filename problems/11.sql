/*
 * Use a JOIN to list all films in the "Family" category.
 * Use table category, film_category, and film.
 */

SELECT
  film.title
FROM category
INNER JOIN film_category ON film_category.category_id = category.category_id
INNER JOIN film ON film.film_id = film_category.film_id
WHERE category.name='Family'
ORDER BY title;
