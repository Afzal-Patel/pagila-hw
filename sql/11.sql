/* 
 * Use a JOIN to list the number of copies of each film in the inventory system that begins with the letter h.
 * Use tables inventory and film.
 */

SELECT
film.film_id
  ,film.title
,COUNT(film.film_id)
FROM inventory
LEFT JOIN film ON inventory.film_id = film.film_id
WHERE film.title ILIKE 'h%'
GROUP BY
  film.film_id
  ,film.title
ORDER BY COUNT(film.film_id) DESC
;
