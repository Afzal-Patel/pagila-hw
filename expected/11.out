 film_id |        title         | count 
---------+----------------------+-------
     412 | HEAVYWEIGHTS BEAST   |     8
     444 | HUSTLER PARTY        |     8
     434 | HORROR REIGN         |     8
     418 | HOBBIT ALIEN         |     8
     403 | HARRY IDAHO          |     8
     408 | HEAD STRANGER        |     7
     397 | HANKY OCTOBER        |     7
     391 | HALF OUTFIELD        |     7
     409 | HEARTBREAKERS BRIGHT |     7
     443 | HURRICANE AFFAIR     |     7
     402 | HARPER DYING         |     6
     392 | HALL CASSIDY         |     6
     429 | HONEY TIES           |     6
     436 | HOURS RAGE           |     6
     414 | HELLFIGHTERS SIERRA  |     6
     439 | HUNCHBACK IMPOSSIBLE |     6
     428 | HOMICIDE PEACH       |     6
     417 | HILLS NEIGHBORS      |     6
     395 | HANDICAP BOONDOCK    |     6
     416 | HIGHBALL POTTER      |     6
     406 | HAUNTING PIANIST     |     6
     415 | HIGH ENCINO          |     6
     433 | HORN WORKING         |     6
     437 | HOUSE DYNAMITE       |     6
     445 | HYDE DOCTOR          |     6
     438 | HUMAN GRAFFITI       |     5
     410 | HEAVEN FREEDOM       |     5
     432 | HOPE TOOTSIE         |     5
     424 | HOLOCAUST HIGHBALL   |     5
     420 | HOLES BRANNIGAN      |     5
     446 | HYSTERICAL GRAIL     |     4
     427 | HOMEWARD CIDER       |     4
     394 | HAMLET WISDOM        |     4
     396 | HANGING DEEP         |     4
     423 | HOLLYWOOD ANONYMOUS  |     4
     440 | HUNGER ROOF          |     4
     421 | HOLIDAY GAMES        |     4
     430 | HOOK CHARIOTS        |     4
     398 | HANOVER GALAXY       |     4
     442 | HUNTING MUSKETEERS   |     3
     431 | HOOSIERS BIRDCAGE    |     3
     413 | HEDWIG ALTER         |     3
     426 | HOME PITY            |     3
     441 | HUNTER ALTER         |     2
     399 | HAPPINESS UNITED     |     2
     407 | HAWK CHILL           |     2
     422 | HOLLOW JEOPARDY      |     2
     401 | HAROLD FRENCH        |     2
     425 | HOLY TADPOLE         |     2
     400 | HARDLY ROBBERS       |     2
     393 | HALLOWEEN NUTS       |     2
     411 | HEAVENLY GUN         |     2
     405 | HAUNTED ANTITRUST    |     2
     435 | HOTEL HAPPINESS      |     2
(54 rows)
